packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -742,10 +745,12 @@ function restoreFocusToElement(node: FocusableElement) {
  * Create a [TreeWalker]{@link https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker}
  * that matches all focusable/tabbable elements.
  */
-export function getFocusableTreeWalker(root: Element, opts?: FocusManagerOptions, scope?: Element[]) {
+export function getFocusableTreeWalker(root: Element | ShadowRoot, opts?: FocusManagerOptions, scope?: Element[]) {
   let selector = opts?.tabbable ? TABBABLE_ELEMENT_SELECTOR : FOCUSABLE_ELEMENT_SELECTOR;
-  let walker = getOwnerDocument(root).createTreeWalker(
-    root,
+  let doc = root instanceof ShadowRoot ? root :  (getRootNode(root) || getOwnerDocument(root));
```

Comment:
will all of these `instanceof`'s have issues?

Reply:
Updated in the last commit.